bothan-lib/src/worker/rest.rs

```diff
@@ -41,30 +43,53 @@ pub async fn start_polling<S: Store, E: Display, P: AssetInfoProvider<Error = E>
 
         select! {
             _ = cancellation_token.cancelled() => break,
-            r = timeout(interval.period(), provider.get_asset_info(&ids)) => handle_polling_result(r, &store).await,
+            r = handle_get_asset_info(&provider, &ids, update_interval, meter_name) => handle_polling_result(r, &store, meter_name).await,
         }
     }
 }
 
+async fn handle_get_asset_info<P, E>(
+    provider: &P,
+    ids: &Vec<String>,
```

Comment:
dont use ref to vec, use slice
```suggestion
    ids: &[String],
```